### PR TITLE
Eliminated no-type/no-original TypeTrees.

### PR DESCRIPTION
**Pull request by paulp**
Noticed many calls reaching a spot which they should not.
Added a devWarning so the next guy might find it more quickly.

### CODE REVIEW

**Comment by adriaanm** [comment]
PLS REBUILD ALL
NOLITTER!

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -5241,11 +5241,13 @@ trait Typers extends Adaptations with Tags with TypersTracking {
             case _ => tree
           }
         }
-        else
+        else {
           // we should get here only when something before failed
           // and we try again (@see tryTypedApply). In that case we can assign
           // whatever type to tree; we just have to survive until a real error message is issued.
+          devWarning(tree.pos, s"TypeTree reached typedTypeTree with null original. Assigning Any in desperation.")
           tree setType AnyTpe
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: I wonder what was the motivation for this.

Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: I'm not sure if I understand the question (the original author's motivation?) but as for me, the comment says approximately "compilation has failed, we'll pitch an Any into the mix and an error will be reported." And I see it walking that code path 1500 times during a successful compilation. And when one is trying to make the pattern matcher a tiny bit sound one starts seeing the random introduction of "Any" types as deadly poison.

Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Yeah, the motivation of `tree setType AnyTpe`. And thanks for explaining the background for your change!

**Comment by paulp** [comment]
I eventually caught a glimpse through the noise that this actually doesn't compile. (I take it I'm not the only one for whom that was not self-apparent.) That'll teach me to try to carve a little patch off a big patch.

**Comment by adriaanm** [comment]
That's a bug in the scripts. Looking into it.

**Comment by adriaanm** [comment]
"-e" missing on bash invocation

**Comment by adriaanm** [comment]
Should be fixed by https://github.com/scala/jenkins-scripts/commit/b7d564b976bebcaed55e42ba3bb747e455078f02

**Comment by adriaanm** [comment]
Please reopen. Trying to remove red from queue.

**Comment by paulp** [comment]
Updated with less broken cherrypick.

**Comment by paulp** [comment]
PLS REBUILD ALL
(Not sure if I have to do that)

**Comment by adriaanm** [comment]
No need for an explicit rebuild if the commits are new.

**Comment by adriaanm** [comment]
It looks like something went wrong with your rebase.

**Comment by paulp** [comment]
It's awfully challenging to find the part which suggests what the problem was. I took a stab.

**Pull request closed without merging**